.ci/pipelines/pr-gke.Jenkinsfile

```diff
@@ -20,65 +20,124 @@ pipeline {
     }
 
     stages {
-        stage('Run checks in parallel') {
-            failFast true
-            parallel {
-                stage('Validate Jenkins pipelines') {
-                    when {
-                        expression {
-                            arePipelinesModified()
+        stage('setup-ci-env') {
+            when {
+                expression {
+                    notOnlyDocs()
+                }
+            }
+            stages {
+                stage('build-ci-image') {
+                    steps {
+                        sh 'make -C .ci get-test-artifacts ci-build-image'
+                        sh '.ci/setenvconfig pr'
+                    }
+                }
+                stage("validate-jenkins-pipelines") {
+                    steps {
+                        script {
+                            sh 'make -C .ci TARGET=validate-jenkins-pipelines ci'
                         }
                     }
+                }
+            }
+        }
+        stage('build') {
+            when {
+                expression {
+                    notOnlyDocs()
+                }
+            }
+            failFast true
+            parallel {
+                stage("build-e2e-tests-docker-image") {
                     steps {
-                        sh 'make -C .ci TARGET=validate-jenkins-pipelines ci'
+                        sh 'make -C .ci TARGET=e2e-docker-multiarch-build ci'
```

Comment:
We used to call `e2e-compile` before creating the image to catch compile errors in e2e tests early. I am not sure if the extra time is worth the tradeoff though. 

Also I think we have a few unused Makefile targets now like `setup-e2e` that we could potentially get rid of?

Reply:
Indeed, I removed `e2e-compile` because I wanted to optimize to reduce execution time and like you said I don't think it is worth it. As it is rare to introduce such error, I think we pay the extra time in 99% of the case.
With the latest pipeline layout, compile errors in e2e tests will be catched early because the pipeline starts by building the e2e-tests image.

Reply:
> Also I think we have a few unused Makefile targets now like setup-e2e that we could potentially get rid of?

The other pipelines are still using it. The goal is to split them too and then we can clean up.